### ISSUE TITLE
Error when processing status during migration

### ISSUE THREAD

**Issue by Octogonapus** [ericboehs/z2g-ui#1]
Hello and thanks for this great project. I'm experiencing this error when trying to run a migration:
```
Failed to process status for issue 49: undefined method `[]' for nil
```
This error is printed for every issue, and no issues are migrated into the GitHub project. Do you know what might cause this error?

Some extra info:
Running on Ubuntu 24.04.1
commit `8f2d9b602799451bdf19d1e994a40ab2850db67b`
ruby `3.3.6 (2024-11-05 revision 75015d4c1f) [x86_64-linux]`

The `github_project`, `issue`, and `@github_status_field` objects:
```
I, [2024-12-06T11:23:42.823986 #2114659]  INFO -- : #<Github::Project:0x00007f15f5d831c0 @token="redacted", @organization="redacted", @number=8>
I, [2024-12-06T11:23:42.824017 #2114659]  INFO -- : {"url"=>"redacted", "number"=>"49", "repoName"=>"redacted", "githubStatus"=>"f13d5554", "githubStatusName"=>"Review/QA"}
I, [2024-12-06T11:23:42.824035 #2114659]  INFO -- : {"id"=>"redacted", "name"=>"Status", "options"=>[{"id"=>"redacted", "name"=>"New Issues"}, {"id"=>"redacted", "name"=>"Icebox"}, {"id"=>"redacted", "name"=>"Epics"}, {"id"=>"redacted", "name"=>"Backlog"}, {"id"=>"redacted", "name"=>"In Progress"}, {"id"=>"redacted", "name"=>"Review/QA"}, {"id"=>"redacted", "name"=>"Done"}]}
```

Backtrace:
```
E, [2024-12-06T11:23:43.207850 #2114659] ERROR -- : /home/salmon/git/z2g-ui/github_project.rb:231:in `add_issue'
/home/salmon/git/z2g-ui/app.rb:687:in `block (3 levels) in <class:App>'
/home/salmon/git/z2g-ui/app.rb:681:in `each'
/home/salmon/git/z2g-ui/app.rb:681:in `block (2 levels) in <class:App>'
/home/salmon/git/z2g-ui/app.rb:680:in `each'
/home/salmon/git/z2g-ui/app.rb:680:in `block in <class:App>'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1804:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1804:in `block in compile!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1071:in `block (3 levels) in route!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1089:in `route_eval'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1071:in `block (2 levels) in route!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1120:in `block in process_route'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1118:in `catch'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1118:in `process_route'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1069:in `block in route!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1066:in `each'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1066:in `route!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1190:in `block in dispatch!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1161:in `catch'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1161:in `invoke'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1185:in `dispatch!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1001:in `block in call!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1161:in `catch'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1161:in `invoke'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1001:in `call!'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:990:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-protection-4.0.0/lib/rack/protection/xss_header.rb:20:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-protection-4.0.0/lib/rack/protection/base.rb:53:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-protection-4.0.0/lib/rack/protection/path_traversal.rb:18:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-protection-4.0.0/lib/rack/protection/json_csrf.rb:28:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-protection-4.0.0/lib/rack/protection/base.rb:53:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-protection-4.0.0/lib/rack/protection/base.rb:53:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-protection-4.0.0/lib/rack/protection/frame_options.rb:33:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-session-2.0.0/lib/rack/session/abstract/id.rb:272:in `context'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-session-2.0.0/lib/rack/session/abstract/id.rb:266:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/null_logger.rb:13:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/head.rb:15:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/show_exceptions.rb:23:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:224:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:2115:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1674:in `block in call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1890:in `synchronize'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:1674:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/tempfile_reaper.rb:20:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/lint.rb:66:in `response'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/lint.rb:41:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/show_exceptions.rb:31:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/common_logger.rb:43:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/sinatra-4.0.0/lib/sinatra/base.rb:266:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/rack-3.1.8/lib/rack/content_length.rb:20:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/puma-6.4.3/lib/puma/configuration.rb:272:in `call'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/puma-6.4.3/lib/puma/request.rb:100:in `block in handle_request'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/puma-6.4.3/lib/puma/thread_pool.rb:378:in `with_force_shutdown'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/puma-6.4.3/lib/puma/request.rb:99:in `handle_request'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/puma-6.4.3/lib/puma/server.rb:464:in `process_client'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/puma-6.4.3/lib/puma/server.rb:245:in `block in run'
/home/salmon/.rvm/gems/ruby-3.3.6/gems/puma-6.4.3/lib/puma/thread_pool.rb:155:in `block in spawn_thread'
```

**Comment by ericboehs** [comment]
I'm guessing something to do with your permission for the GitHub token you provided?

I don't maintain this project much but will be happy to accept PRs.

Specifically the problem is on 239. I need to handle `nil` checks there as it seems your issues aren't coming back with a `:resource` or perhaps `:data` key. You could put a debugger/binding.pry in there and inspect the result of the query:

```ruby
id_result = query(
  %Q{query{
    resource(url: "#{issue_url}") {
      ... on Issue {
        id
      }
    }
  }}
)
puts id_result
binding.pry

issue_id = id_query[:data][:resource][:id]
```

**Comment by Octogonapus** [comment]
I followed your instructions of making a classic token with `project` scope. It seems that's insufficient, at least for doing migrations in an organization. I got it to work with a fine-grained token owned by the org with the scopes: read/write projects (organization), read/write issues (all repos), read/write pull requests (all repos), read metadata (all repos). Of course, this also requires changing the token scope validation. I don't know if this is for certain the right way to do it so I'll leave that code alone.
